tests/test_coupons.py

```diff
@@ -122,6 +123,14 @@ def test_delete(self):
             coupon.refresh_from_db()
             self.assertTrue(coupon.is_deleted)
 
+        with mock.patch("aa_stripe.models.StripeCoupon.delete") as mocked_delete:
+            # also test the overriden queryset's delete
+            coupon2 = self._create_coupon(coupon_id="CPON2")
+            coupon3 = self._create_coupon(coupon_id="CPON3")
+            result = StripeCoupon.objects.filter(pk__in=[coupon2.pk, coupon3.pk]).delete()
+            self.assertEqual(mocked_delete.call_count, 2)
+            self.assertEqual(result, (0, {"aa_stripe.StripeCoupon": 0}))
```

Comment:
Please also add test for StripeCoupon.objects.count() after the delete.

Reply:
Added in the CR commit.